protocol/messenger_contact_verification.go

```diff
@@ -723,19 +723,53 @@ func (m *Messenger) MarkAsUntrustworthy(ctx context.Context, contactID string) e
 	return m.SyncTrustedUser(ctx, contactID, verification.TrustStatusUNTRUSTWORTHY, m.dispatchMessage)
 }
 
-func (m *Messenger) RemoveTrustStatus(ctx context.Context, contactID string) error {
+func (m *Messenger) RemoveTrustStatus(ctx context.Context, contactID string) (*MessengerResponse, error) {
+	contact, ok := m.allContacts.Load(contactID)
+	if !ok {
+		return nil, errors.New("Contact not found")
+	}
+	currentTrustStatus := contact.TrustStatus
+
 	err := m.verificationDatabase.SetTrustStatus(contactID, verification.TrustStatusUNKNOWN, m.getTimesource().GetCurrentTime())
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	err = m.SyncTrustedUser(ctx, contactID, verification.TrustStatusUNKNOWN, m.dispatchMessage)
+	if err != nil {
+		return nil, err
+	}
+
+	contact.TrustStatus = verification.TrustStatusUNKNOWN
+
+	// drop verification status
+	if currentTrustStatus != verification.TrustStatusUNTRUSTWORTHY {
+		contact.VerificationStatus = VerificationStatusUNVERIFIED
+		contact.LastUpdatedLocally = m.getTimesource().GetCurrentTime()
+	}
+
+	// Save the updated contact
+	err = m.persistence.SaveContact(contact, nil)
```

Comment:
i suggest to save & sync contact only if you update VerificationStatus and LastUpdatedLocally. And move `SetTrustStatus` and `SyncTrustedUser` to else